loom/script/native/lsNativeInterface.cpp

```diff
@@ -409,6 +411,32 @@ void *lualoom_getnativepointer(lua_State *L, int index, bool replaceIndex, const
     return pointer;
 }
 
+#if 0
+// Dump the managed table, filtering to only show two types.
+static void lualoom_dumpmanagedtable(lua_State *L, Type *filterA, Type *filterB)
+{
+    printf("----- managed table dump -----\n");
+
+    NativeTypeBase *ntbA = NativeInterface::getNativeType(filterA);
+    NativeTypeBase *ntbB = NativeInterface::getNativeType(filterB);
+
+    lua_rawgeti(L, LUA_GLOBALSINDEX, LSINDEXMANAGEDNATIVESCRIPT);
+    lua_pushnil(L);  // first key
+    while (lua_next(L, -2) != 0)
+    {
+        Detail::UserdataPtr *ud = (Detail::UserdataPtr *)lua_topointer(L, -2);
+        if(ud->m_nativeType == ntbA || ud->m_nativeType == ntbB)
+        {
+            // uses 'key' (at index -2) and 'value' (at index -1)
+            printf("%x - %s *%x\n", ud, ud->m_nativeType->getCTypeName().c_str(), ud->getPointer());
+        }
+
+        // removes 'value'; keeps 'key' for next iteration
+        lua_pop(L, 1);
+    }
+    lua_pop(L, 1);
+}
+#endif
```

Comment:
Useful debug code. I've cleaned this up to be better documented and also clarify why the calls to it are commented out but kept.